astropy/coordinates/jparser.py

```diff
@@ -0,0 +1,81 @@
+import re
+
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+
+def _sexagesimal(g):
+    # convert matched regex groups to sexigesimal array
+    sign, h, m, s, frac = g
+    sign = [1, -1][sign == '-']
+    s = '.'.join((s, frac))
+    return sign * np.array([h, m, s], float)
+
+
+class Jparser(object):
+    """
+    Class that extracts coordinates from object names containing coordinated
+    eg: '2MASS J06495091-0737408'
+    """
+    basePattern = '([+-]?)(\d{1,2})(\d{2})(\d{2})\.?(\d{0,3})'
```

Comment:
I think you need to be more careful with the regex's.  E.g. this would match `J+012345012345678`.